src/Operation/Count.php

```diff
@@ -183,7 +181,7 @@ public function getCommandDocument(Server $server)
      *
      * @return array
      */
-    private function createCommandDocument()
+    private function createCommandDocument(): array
```

Comment:
I noticed `getCommandDocument()` above also has no return type hint. If this is due to the Explainable interface not using the return type hint (likely a BC break even though we define the class as `@internal`), I think we'd still be permitted to use `array` in implementing classes per covariance rules. If so, you may want to just make this change en masse.

Reply:
The `Explainable` interface not having a return type is no problem, as covariance would allow us to add it. However, despite the interface being marked as `@internal`, the classes implementing the interface are not, and adding the return type to the interface would require adding it to the implementing classes as well. This again would constitute a BC break.

I'm not sure how the `@internal` is intended to be read: do we not expect users to implement the interface in their own classes, or do we treat the entire contract as internal for users not to rely on? If it's the latter, I would suggest fixing this in 2.0 by making the method `final` in implementing classes.

Reply:
I don't think there's any rules for `@internal`. It was a convention from the [old phpDocumentor days](https://pear.php.net/package/PhpDocumentor/docs/latest/phpDocumentor/tutorial_tags.internal.pkg.html).

PHPUnit goes a step further to clarify this with:

```
@internal This class is not covered by the backward compatibility promise for PHPUnit
```

The idea for PHPLIB was to allow us flexibility to ignore BC breaks on `@internal` functions/classes, but I dropped the ball by never considering BC breaks could still happen through the implementing classes. Granted, this was also before return type hints were usable for us.

This is all a good learning experience that should form the basis for a 2.0 ticket.